### DISCUSSION TITLE
How to deal with large amount of strokes

### DISCUSSION THREAD

**Discussion by rnike** [pixijs/pixijs#10483] [General]
I am creating an application that can draw on the stage, I use Graphics for each stroke, but the performance drops when the stroke increases.

Something I do is like:

```js
const addDrawing = (points, style) => {
  const g = new Graphics();

  g.lineStyle(...style);

  for (let i = 0; I < points.length; i += 2) {
    const x = points[i];
    const y = points[i + 1];

    if (i === 0) {
      g.moveTo(x, y);
    } else {
      g.lineTo(x, y);
    }
  }

  stage.addChild(g);
};
```

The FPS drops a lot after adding hundreds of drawings and moving around on stage.

I looked into the Chrome dev tool and saw the GPU takes a lot of time to update the stage.

I tried cache as bitmap, and it did increase the speed of rendering, but it seems not a good idea to cache hundreds or thousands of different drawings in the memory.

Is there another way to solve the issue?

**Comment by rnike** [comment]
I ended up using SimpleRope and simplify-js for this case the performance is pretty well

**Reply by Amansaxena001** [reply]
Is it possible to use it in a case where we have many graphics objects and these are dynamic that is we can expand or collpase them ( change the dimensions etc) @rnike
